frontend/frontend/honeylogger/__init__.py

```diff
@@ -25,8 +25,11 @@ class Session(Protocol):
     """Representation of an attacker's session while being connected to the honeypot."""
 
     @abstractmethod
-    def begin(self) -> None:
-        """Begins the logging session"""
+    def begin(self, ssh_version: str) -> None:
```

Comment:
This assumes every Session is using SSH. Could we maybe pass in the SSH remote version to honeylogger.create_ssh_session and then to the contstructor of SSHSession instead since it is SSH specific?